### PR TITLE
rust: add support for id and secret args

### PR DESCRIPTION
**Pull request by Esgrove**
Add support for giving IAM id and secret to match the Python vault.

### CODE REVIEW

**Comment by Esgrove** [comment]
> Is this PR trying to add the iam support, or doing some cleanup to code or reordering/changing some arguments to other places? Reviewing this pr makes no sense when there are 100s of irrelevant lines in the diff. Please move the formatting stuff to a separate PR to ease reviewing.

Updated the PR to only contain changes directly related to the new argument options

File: python-pyo3/python/n_vault/vault.py [hunk 1]
```diff
@@ -112,6 +118,8 @@ def delete_many(self, names: Collection[str]) -> None:
             key=self.vault_key,
             prefix=self.vault_prefix,
             profile=self.profile,
+            iam_id=self.vault_iam_id,
```
Comment (kahlstrm) [inline python-pyo3/python/n_vault/vault.py:121]: I wonder if there would some better method passing these values as some combined struct/class but might not be🤷. Seems quite verbose😅. This is just an observation, nothing else.

**Review by kahlstrm — APPROVED** [review]
Looks good

File: python-pyo3/python/n_vault/vault.py (hunk 1 above)
Comment (Esgrove) [reply python-pyo3/python/n_vault/vault.py:121]: Yes, I was thinking of making a `VaultConfig` etc dataclass that would contain the settings but left it like this so the interface stays the same as before. Could add a separate factory method for the rust vault that takes a single config object, will try it out later today.

**Pull request merged**